### PR TITLE
Rename whitelist & blacklist #71

### PR DESCRIPTION
**Pull request by HolyFalafel**
Rename whitelist and blacklist to allowlist and blocklist

### CODE REVIEW

File: tests/transformers/tests/test_modeling_common.py [hunk 1]
```diff
@@ -1883,8 +1883,8 @@ def test_multi_gpu_data_parallel_forward(self):
 
         # some params shouldn't be scattered by nn.DataParallel
         # so just remove them if they are present.
-        blacklist_non_batched_params = ["head_mask", "decoder_head_mask", "cross_attn_head_mask"]
-        for k in blacklist_non_batched_params:
+        blocklist_non_batched_params = ["head_mask", "decoder_head_mask", "cross_attn_head_mask"]
```
Comment (libinta) [inline tests/transformers/tests/test_modeling_common.py:1886]: why do we make this change, is it related?

File: tests/transformers/tests/test_modeling_common.py [hunk 2]
```diff
@@ -1883,8 +1883,8 @@ def test_multi_gpu_data_parallel_forward(self):
...
+        for k in blocklist_non_batched_params:
```
Comment (libinta) [inline tests/transformers/tests/test_modeling_common.py:1887]: same question

File: tests/transformers/tests/test_modeling_common.py (hunk 1 above)
Comment (HolyFalafel) [reply tests/transformers/tests/test_modeling_common.py:1886]: Not related. just another offending code

File: tests/transformers/tests/test_modeling_common.py (hunk 2 above)
Comment (HolyFalafel) [reply tests/transformers/tests/test_modeling_common.py:1887]: same answer

**Review by libinta — APPROVED** [review]

**Review by regisss — APPROVED** [review]
LGTM!

Is it compatible with Synapse v1.14 or should we wait for the release of v1.15 to merge it?

**Comment by HolyFalafel** [comment]
> LGTM!
> 
> Is it compatible with Synapse v1.14 or should we wait for the release of v1.15 to merge it?

This is relevant to a change in v1.15, so it's incompatible for v1.14

**Pull request merged**